snooty/rstspec.toml

```diff
@@ -463,6 +463,43 @@ example = """.. code:: ${1:language}
    ${0:code content}
 """
 
+[directive.io-code-block]
+help = """A block of code and its corresponding output to show the user."""
+argument_type = "string"
+content_type = "block"
+options.caption = "string"
+options.copyable = "boolean"
+options.class = "string"
+options.source = "uri"
+example = """.. io-code-block:: ${1:language}
+   ${2::copyable: (bool)
+   :caption: (string)
+   :class: (string)}
+
+   ${0:code input/code output content}
+"""
+
+[directive.input]
+help = """The code content."""
+argument_type = ["path", "raw"]
+content_type = "raw"
+options.emphasize-lines = "linenos"
+options.linenos = "flag"
+example = """.. io-code-block:: ${1:language}
+   .. input:: ${0:code output or </path/to/file>}
+      :emphasize-lines: (string)
+      :linenos: (flag)
+"""
+
+[directive.output]
+help = """The code output."""
+inherit = "input"
+example = """.. io-code-block:: ${1:language}
+   .. output:: ${0:code output or </path/to/file>}
+      :emphasize-lines: (string)
+      :linenos: (flag)
+"""
+
```

Comment:
I'm not sure why this didn't occur to me previously - I'm sorry. Is there a context in which we'd want the input to be one language and the output another?

/cc @terakilobyte, who is particularly enthused about this feature to weigh in

Reply:
I assumed that the output would just be plaintext, but if we wanted writers to be able to specify a language for the output codeblock it shouldn't be hard.

Reply:
I envision output to mainly be either plaintext or json, but it could be other formats. A Query API to SQL demonstration, or if MongoDB ever creates generators, or if we needed to go into deeper detail on some builders the drivers expose.

Reply:
So would we like the input/output directives to be able to specify a language instead of the `io-code-block `directive specifying the overall language?